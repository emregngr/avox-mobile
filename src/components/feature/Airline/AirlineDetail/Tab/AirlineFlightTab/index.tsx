import React, { useMemo } from 'react'
import { View } from 'react-native'

import { Alliance } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/Alliance'
import { DestinationList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/DestinationList'
import { RoutesList } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airline/AirlineDetail/Tab/AirlineFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { Airline } from '@/types/feature/airline'

interface AirlineFlightTabProps {
  airlineData: Airline
}

export const AirlineFlightTab = ({ airlineData }: AirlineFlightTabProps) => {
  const { selectedTheme } = useThemeStore()
  const { selectedLocale } = useLocaleStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { network, operations } = airlineData ?? {}
  const { alliance } = operations ?? {}
  const {
    destinationCount,
    destinationCountries,
    destinations,
    domesticConnections,
    internationalConnections,
    routes,
  } = network ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const iconColor = useMemo(() => colors.onPrimary100, [colors.onPrimary100])

  const localeStrings = useMemo(
    () => ({
      activeRoutes: getLocale('activeRoutes'),
      alliance: getLocale('alliance'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      mainDestinations: getLocale('mainDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={iconColor}
        stats={memoizedStats}
      />

      <DestinationList
        destinations={destinations}
        iconColor={iconColor}
        title={localeStrings.mainDestinations}
      />

      <RoutesList iconColor={iconColor} routes={routes} title={localeStrings.activeRoutes} />

      {alliance !== 'none' && (
        <Alliance alliance={alliance} iconColor={iconColor} title={localeStrings.alliance} />
      )}
    </View>
  )
}
