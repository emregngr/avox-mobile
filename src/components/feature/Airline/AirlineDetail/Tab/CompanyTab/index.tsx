import React from 'react'
import { View } from 'react-native'

import { Company } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Company'
import { Contact } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Contact'
import { Hub } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Hub'
import { Map } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/Map'
import { SocialMedia } from '@/components/feature/Airline/AirlineDetail/Tab/CompanyTab/Sections/SocialMedia'
import type { Airline } from '@/types/feature/airline'

interface CompanyTabProps {
  airlineData: Airline
}

export const CompanyTab = ({ airlineData }: CompanyTabProps) => {
  const { companyInfo, operations } = airlineData

  return (
    <View className="px-4 gap-y-4">
      <Company airlineData={airlineData} />

      <Contact companyInfo={companyInfo} operations={operations} />

      <SocialMedia companyInfo={companyInfo} />

      <Hub operations={operations} />

      <Map airlineData={airlineData} />
    </View>
  )
}
