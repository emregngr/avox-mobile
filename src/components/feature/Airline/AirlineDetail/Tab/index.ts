export * from './AirlineFlightTab'
export * from './CompanyTab'
export * from './FleetTab'
export * from './SafetyEnvTab'
