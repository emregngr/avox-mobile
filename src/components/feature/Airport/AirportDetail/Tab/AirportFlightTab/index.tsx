import { useMemo } from 'react'
import { View } from 'react-native'

import { AirlinesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/AirlinesList'
import { RoutesList } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/RoutesList'
import { StatsGrid } from '@/components/feature/Airport/AirportDetail/Tab/AirportFlightTab/Sections/StatsGrid'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import useThemeStore from '@/store/theme'
import { themeColors } from '@/themes'
import type { AirportType } from '@/types/feature/airport'

interface AirportFlightTabProps {
  airportData: AirportType
}

export const AirportFlightTab = ({ airportData }: AirportFlightTabProps) => {
  const { selectedLocale } = useLocaleStore()
  const { selectedTheme } = useThemeStore()

  const colors = useMemo(() => themeColors?.[selectedTheme], [selectedTheme])

  const { flightOperations } = airportData ?? {}
  const {
    airlines,
    destinationCount,
    destinationCountries,
    domesticConnections,
    internationalConnections,
    routes,
  } = flightOperations ?? {}

  const memoizedStats = useMemo(
    () => ({
      destinationCount,
      destinationCountries,
      domesticConnections,
      internationalConnections,
    }),
    [destinationCount, destinationCountries, domesticConnections, internationalConnections],
  )

  const localeStrings = useMemo(
    () => ({
      airlines: getLocale('airlines'),
      country: getLocale('country'),
      domesticDestinations: getLocale('domesticDestinations'),
      internationalDestination: getLocale('internationalDestination'),
      popularDestinations: getLocale('popularDestinations'),
      totalDestination: getLocale('totalDestination'),
    }),
    [selectedLocale],
  )

  return (
    <View className="px-4 gap-y-4">
      <StatsGrid
        localeStrings={{
          country: localeStrings.country,
          domesticDestinations: localeStrings.domesticDestinations,
          internationalDestination: localeStrings.internationalDestination,
          totalDestination: localeStrings.totalDestination,
        }}
        iconColor={colors?.onPrimary100}
        stats={memoizedStats}
      />

      <AirlinesList
        airlines={airlines}
        iconColor={colors?.onPrimary100}
        title={localeStrings.airlines}
      />

      <RoutesList
        iconColor={colors?.onPrimary100}
        routes={routes}
        title={localeStrings.popularDestinations}
      />
    </View>
  )
}
