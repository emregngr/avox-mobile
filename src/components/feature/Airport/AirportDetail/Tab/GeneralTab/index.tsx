import React from 'react'
import { View } from 'react-native'

import { Contact } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Contact'
import { General } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/General'
import { Location } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Location'
import { Map } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Map'
import { Services } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/Services'
import { SocialMedia } from '@/components/feature/Airport/AirportDetail/Tab/GeneralTab/Sections/SocialMedia'
import type { Airport } from '@/types/feature/airport'

interface GeneralTabProps {
  airportData: Airport
}

export const GeneralTab = ({ airportData }: GeneralTabProps) => {
  const { airportInfo, facilities, operations } = airportData ?? {}

  return (
    <View className="px-4 gap-y-4">
      <General airportInfo={airportInfo} operations={operations} />

      <Services facilities={facilities} />

      <Contact airportInfo={airportInfo} />

      <SocialMedia airportInfo={airportInfo} />

      <Location operations={operations} />

      <Map airportData={airportData} />
    </View>
  )
}
