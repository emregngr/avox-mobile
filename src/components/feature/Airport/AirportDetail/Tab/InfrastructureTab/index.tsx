import React from 'react'
import { View } from 'react-native'

import { Cargo } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Cargo'
import { Certifications } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Certifications'
import { Facilities } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Facilities'
import { Runway } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Runway'
import { Terminal } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Terminal'
import type { Airport } from '@/types/feature/airport'

interface InfrastructureTabProps {
  airportData: Airport
}

export const InfrastructureTab = ({ airportData }: InfrastructureTabProps) => {
  const { cargo, facilities, infrastructure, safety } = airportData || {}

  return (
    <View className="px-4 gap-y-4">
      <Terminal infrastructure={infrastructure} />

      <Runway infrastructure={infrastructure} />

      <Facilities facilities={facilities} infrastructure={infrastructure} />

      <Cargo cargo={cargo} />

      <Certifications safety={safety} />
    </View>
  )
}
