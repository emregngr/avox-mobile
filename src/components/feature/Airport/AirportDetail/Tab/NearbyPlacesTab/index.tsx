import React, { useCallback, useMemo } from 'react'
import { View } from 'react-native'

import { AttractionCard } from '@/components/feature/Airport/AirportDetail/Tab/NearbyPlacesTab/Cards/AttractionCard'
import { useMapNavigation } from '@/hooks/maps/useMapNavigation'
import { getLocale } from '@/locales/i18next'
import useLocaleStore from '@/store/locale'
import type { Airport } from '@/types/feature/airport'

interface NearbyPlacesTabProps {
  airportData: Airport
}
interface AttractionCard {
  attractionId: number
  attractionLatitude: number
  attractionLongitude: number
  attractionName: string
  description: string
  formattedDistance: string
  handleDirectionPress: () => void
}

export const NearbyPlacesTab = ({ airportData }: NearbyPlacesTabProps) => {
  const { selectedLocale } = useLocaleStore()

  const { nearbyAttractions } = airportData || {}

  const localeStrings = useMemo(
    () => ({
      getDirection: getLocale('getDirection'),
      km: getLocale('km'),
    }),
    [selectedLocale],
  )

  const mapLocaleStrings = useMemo(
    () => ({
      appleMaps: getLocale('appleMaps'),
      cancel: getLocale('cancel'),
      googleMaps: getLocale('googleMaps'),
      selectMapApp: getLocale('selectMapApp'),
    }),
    [selectedLocale],
  )

  const { openMapNavigation } = useMapNavigation(mapLocaleStrings)

  const attractionItems: AttractionCard[] = useMemo(
    () =>
      nearbyAttractions?.map(attraction => {
        const {
          attractionCoordinates: { attractionLatitude, attractionLongitude },
          attractionId,
          attractionName,
          description,
          distanceKm,
        } = attraction

        const handleDirectionPress = () => {
          openMapNavigation(attractionLatitude, attractionLongitude, attractionName)
        }

        const formattedDistance = `${distanceKm} ${localeStrings.km}`

        return {
          attractionId,
          attractionLatitude,
          attractionLongitude,
          attractionName,
          description,
          formattedDistance,
          handleDirectionPress,
        }
      }) || [],
    [nearbyAttractions, openMapNavigation, localeStrings.km],
  )

  const renderAttraction = useCallback(
    (item: AttractionCard) => (
      <AttractionCard
        attractionId={item.attractionId}
        attractionName={item.attractionName}
        description={item.description}
        formattedDistance={item.formattedDistance}
        getDirectionText={localeStrings.getDirection}
        handleDirectionPress={item.handleDirectionPress}
        key={item.attractionId}
      />
    ),
    [localeStrings.getDirection],
  )

  return <View className="px-4 gap-y-4">{attractionItems?.map(renderAttraction)}</View>
}
