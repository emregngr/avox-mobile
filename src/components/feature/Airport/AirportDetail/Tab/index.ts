export * from './AirportFlightTab'
export * from './GeneralTab'
export * from './InfrastructureTab'
export * from './NearbyPlacesTab'
