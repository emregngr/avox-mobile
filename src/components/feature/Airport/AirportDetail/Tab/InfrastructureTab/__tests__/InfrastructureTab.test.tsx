import { render } from '@testing-library/react-native'
import React from 'react'

import { InfrastructureTab } from '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab'

const mockedTerminal = jest.fn()
const mockedRunway = jest.fn()
const mockedFacilities = jest.fn()
const mockedCargo = jest.fn()
const mockedCertifications = jest.fn()

jest.mock(
  '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Terminal',
  () => {
    const { Text } = require('react-native')
    return {
      Terminal: (props: any) => {
        mockedTerminal(props)
        return <Text testID="terminal-section">Terminal</Text>
      },
    }
  },
)

jest.mock(
  '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Runway',
  () => {
    const { Text } = require('react-native')
    return {
      Runway: (props: any) => {
        mockedRunway(props)
        return <Text testID="runway-section">Runway</Text>
      },
    }
  },
)

jest.mock(
  '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Facilities',
  () => {
    const { Text } = require('react-native')
    return {
      Facilities: (props: any) => {
        mockedFacilities(props)
        return <Text testID="facilities-section">Facilities</Text>
      },
    }
  },
)

jest.mock('@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Cargo', () => {
  const { Text } = require('react-native')
  return {
    Cargo: (props: any) => {
      mockedCargo(props)
      return <Text testID="cargo-section">Cargo</Text>
    },
  }
})

jest.mock(
  '@/components/feature/Airport/AirportDetail/Tab/InfrastructureTab/Sections/Certifications',
  () => {
    const { Text } = require('react-native')
    return {
      Certifications: (props: any) => {
        mockedCertifications(props)
        return <Text testID="certifications-section">Certifications</Text>
      },
    }
  },
)

const mockedAirportData: any = {
  cargo: {
    capacity: 5_500_000,
    area: 1_400_000,
  },
  facilities: {
    lounges: 12,
    terminals: 1,
  },
  infrastructure: {
    runways: [{ id: '1', length: 4100 }],
    terminals: [{ id: 'T1', area: 1_440_000 }],
  },
  safety: {
    certifications: ['ISO 9001', 'ISO 14001'],
    fireFightingCategory: 'CAT 10',
  },
}

describe('InfrastructureTab Component', () => {
  it('should render all section components', () => {
    const { getByTestId } = render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(getByTestId('terminal-section')).toBeTruthy()
    expect(getByTestId('runway-section')).toBeTruthy()
    expect(getByTestId('facilities-section')).toBeTruthy()
    expect(getByTestId('cargo-section')).toBeTruthy()
    expect(getByTestId('certifications-section')).toBeTruthy()
  })

  it('should pass correct props to Terminal component', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedTerminal).toHaveBeenCalledWith({
      infrastructure: mockedAirportData.infrastructure,
    })
  })

  it('should pass correct props to Runway component', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedRunway).toHaveBeenCalledWith({
      infrastructure: mockedAirportData.infrastructure,
    })
  })

  it('should pass correct props to Facilities component', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedFacilities).toHaveBeenCalledWith({
      facilities: mockedAirportData.facilities,
      infrastructure: mockedAirportData.infrastructure,
    })
  })

  it('should pass correct props to Cargo component', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedCargo).toHaveBeenCalledWith({
      cargo: mockedAirportData.cargo,
    })
  })

  it('should pass correct props to Certifications component', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedCertifications).toHaveBeenCalledWith({
      safety: mockedAirportData.safety,
    })
  })

  it('should handle undefined airportData gracefully', () => {
    render(<InfrastructureTab airportData={undefined as any} />)

    expect(mockedTerminal).toHaveBeenCalledWith({ infrastructure: undefined })
    expect(mockedFacilities).toHaveBeenCalledWith({
      facilities: undefined,
      infrastructure: undefined,
    })
    expect(mockedCargo).toHaveBeenCalledWith({ cargo: undefined })
    expect(mockedCertifications).toHaveBeenCalledWith({ safety: undefined })
  })

  it('should handle partial airportData (missing infrastructure)', () => {
    const partialData: any = {
      cargo: mockedAirportData.cargo,
      safety: mockedAirportData.safety,
    }

    render(<InfrastructureTab airportData={partialData} />)

    expect(mockedTerminal).toHaveBeenCalledWith({ infrastructure: undefined })
    expect(mockedRunway).toHaveBeenCalledWith({ infrastructure: undefined })
    expect(mockedCargo).toHaveBeenCalledWith({ cargo: partialData.cargo })
  })

  it('should handle partial airportData (missing safety and cargo)', () => {
    const partialData: any = {
      facilities: mockedAirportData.facilities,
      infrastructure: mockedAirportData.infrastructure,
    }

    render(<InfrastructureTab airportData={partialData} />)

    expect(mockedFacilities).toHaveBeenCalledWith({
      facilities: partialData.facilities,
      infrastructure: partialData.infrastructure,
    })
    expect(mockedCargo).toHaveBeenCalledWith({ cargo: undefined })
    expect(mockedCertifications).toHaveBeenCalledWith({ safety: undefined })
  })

  it('should call all section components exactly once', () => {
    render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(mockedTerminal).toHaveBeenCalledTimes(1)
    expect(mockedRunway).toHaveBeenCalledTimes(1)
    expect(mockedFacilities).toHaveBeenCalledTimes(1)
    expect(mockedCargo).toHaveBeenCalledTimes(1)
    expect(mockedCertifications).toHaveBeenCalledTimes(1)
  })
})

describe('InfrastructureTab Component Snapshot', () => {
  it('should render the InfrastructureTab Component successfully', () => {
    const { toJSON } = render(<InfrastructureTab airportData={mockedAirportData} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
