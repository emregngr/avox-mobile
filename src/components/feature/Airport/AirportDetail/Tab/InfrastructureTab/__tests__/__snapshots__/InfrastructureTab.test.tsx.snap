// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InfrastructureTab Component Snapshot should render the InfrastructureTab Component successfully 1`] = `
<View
  className="px-4 gap-y-4"
>
  <Text
    testID="terminal-section"
  >
    Terminal
  </Text>
  <Text
    testID="runway-section"
  >
    Runway
  </Text>
  <Text
    testID="facilities-section"
  >
    Facilities
  </Text>
  <Text
    testID="cargo-section"
  >
    Cargo
  </Text>
  <Text
    testID="certifications-section"
  >
    Certifications
  </Text>
</View>
`;
